File: backend/entity/AssetType.go
package entity

import (
    "gorm.io/gorm"
    "time"
)

type AssetType struct {
    gorm.Model
    Name       string    `json:"Name"`
    Type       string    `json:"Type"`
    PenaltyFee float64   `json:"PenaltyFee"`
    Date       time.Time `json:"Date"`

    // Relations
    RoomAssets []RoomAsset `json:"RoomAssets" gorm:"foreignKey:AssetTypeID"`
}

File: frontend/src/interfaces/AssetType.ts
  import type { RoomAsset } from "./RoomAsset";

  export interface AssetType {
    ID?: number;
    CreatedAt?: string;
    UpdatedAt?: string;
    DeletedAt?: string | null;

    Name: string;
    Type:string ;
    PenaltyFee: number;
    Date: string; // ISO date string

    // Relations
    RoomAssets?: RoomAsset[];
  }
